lib/refocus/samples/collector.rb

```diff
@@ -24,6 +24,16 @@ def upsert_bulk
       end
       alias_method :submit, :upsert_bulk
 
+      def check_status(upsert_bulk_response)
+        job_id = upsert_bulk_response["jobId"]
+        if job_id
+          json(http.get("upsert/bulk/#{job_id}/status"))
+        else
+          warn "The jobId was not found in the previous upsert_bulk_response - #{upsert_bulk_response} - Status cannot be checked."
+          "{}" # return empty json
+        end
+      end
```

Comment:
@kkim-sf 

I'd suggest passing the actual `jobId` value to this method, instead of the entire hash. The rest of the data in the hash isn't really important to this method; I think we should pass the bare minimum of data that is necessary for this method to do its job. Then we won't need to check for the presence of the job id, except maybe with something like this:

```ruby
def check_status(job_id) {
  raise ArgumentError, "job_id cannot be nil" if job_id.nil?
  json(http.get("upsert/bulk/#{job_id}/status"))
}
```

Reply:
@sheax0r I can go that route, however, then we're asking the users of the gem to know to parse out the job id or having a separate method to do so, which would then just be an additional step to what's implemented. I think the point of using these gems is to be able to easily use the api -- and to me that sort of defeats the purpose.

Reply:
plus the response hash is really just -
{"status":OK, "jobId":1234}

Reply:
@kkim-sf asking the implementor to understand the hash they get back seems reasonable to me. Yes, we want to make it easy to use the API, but I also want to present an interface that looks like the API itself. I want this gem to be a fairly lightweight wrapper around the API.

We *could* implement models around request and response objects, as well as error types. However, the more we do this, the more we end up hiding from the end user. This means two things:

1. The library will end up containing logic and if the API ends up changing, the library will need to change with it or stop working.
2. We might end up implementing logic that isn't exactly what is desired by a subset of the end users, and then they'll need to work around the library, rather than *with* it.

Because of this, I prefer to keep the interfaces to the gem methods similar to the interfaces to the underlying API methods. Making more formal wrappers around these objects aren't necessarily wrong or bad, it's just not what I want this library to do.